dbms/src/Server/BgStorageInit.cpp

```diff
@@ -66,7 +69,26 @@ void BgStorageInitHolder::start(Context & global_context, const LoggerPtr & log,
                 err_cnt++;
                 tryLogCurrentException(log, fmt::format("Storage inited fail, keyspace_id={} table_id={}", ks_id, table_id));
             }
+        };
+
+        size_t default_num_threads = std::max(4UL, std::thread::hardware_concurrency()) * global_context.getSettingsRef().init_thread_count_scale;
+        auto init_storages_thread_pool = ThreadPool(default_num_threads, default_num_threads / 2, default_num_threads * 2);
```

Comment:
This causes `ThreadPool` to have at least `default_num_threads / 2` number of threads, which is a waste because normally we won't have so many tables to load. Better to use a smaller default value for maxFreeThreads.

Reply:
A large max_free_thread only will hold a lot of free thread when there have a lot of tasks first and then have only small number of tasks later. While, the threadpool here is just a temporary thread pool, and will be released soon. So I think a large max_free_thread will not have a bad affect here.